src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -175,11 +175,261 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Bindin
                 UnaryPatternSyntax p => BindUnaryPattern(p, inputType, inputValEscape, hasErrors, diagnostics, underIsPattern),
                 RelationalPatternSyntax p => BindRelationalPattern(p, inputType, hasErrors, diagnostics),
                 TypePatternSyntax p => BindTypePattern(p, inputType, hasErrors, diagnostics),
+                SlicePatternSyntax p => BindSlicePattern(p, inputType, permitDesignations, ref hasErrors, misplaced: true, diagnostics),
                 _ => throw ExceptionUtilities.UnexpectedValue(node.Kind()),
             };
         }
 
-        private BoundPattern BindDiscardPattern(DiscardPatternSyntax node, TypeSymbol inputType)
+        private BoundPattern BindSlicePattern(
+            SlicePatternSyntax node,
+            TypeSymbol inputType,
+            bool permitDesignations,
+            ref bool hasErrors,
+            bool misplaced,
+            BindingDiagnosticBag diagnostics)
+        {
+            if (misplaced && !hasErrors)
+            {
+                diagnostics.Add(ErrorCode.ERR_MisplacedSlicePattern, node.Location);
+                hasErrors = true;
+            }
+
+            IndexerArgumentInfo? info = null;
+
+            TypeSymbol sliceType;
+            if (inputType.IsSZArray())
+            {
+                sliceType = inputType;
+            }
+            else if (TryFindRangeIndexerPatternForListPattern(node, inputType, out info, diagnostics))
+            {
+                sliceType = info.Symbol.GetTypeOrReturnType().Type;
+            }
+            else
+            {
+                // PROTOTYPE(list-patterns) Cover any other type that we intend to cover, multi-dimensional arrays?
+                if (!inputType.IsErrorType())
+                    diagnostics.Add(ErrorCode.ERR_UnsupportedTypeForSlicePattern, node.Location, inputType.ToDisplayString());
+                hasErrors = true;
+                sliceType = CreateErrorType();
+            }
+
+            BoundPattern? pattern = node.Pattern is not null
+                ? BindPattern(node.Pattern, sliceType, ExternalScope, permitDesignations, hasErrors, diagnostics)
+                : null;
+
+            return new BoundSlicePattern(node, sliceType: sliceType, pattern, info, inputType: inputType, narrowedType: inputType, hasErrors);
+        }
+
+        private ImmutableArray<BoundPattern> BindListPatternSubpatterns(
+            PropertyPatternClauseSyntax node,
```

Comment:
Consider passing in `node.Subpatterns` instead, since that's the only need we have for `node` in this method and that would align better with the purpose of this method.